cmd/icingadb/main.go

```diff
@@ -33,47 +35,76 @@ func main() {
 
 func run() int {
 	cmd := command.New()
+	// When started by systemd, NOTIFY_SOCKET is set by systemd for Type=notify supervised services, which is the
+	// default setting for the Icinga DB service. So we notify that Icinga DB finished starting up and also configure
+	// logging into the systemd journal instead of stderr.
+	// Always add this line before output assignment
+	_, _ = daemon.SdNotify(false, daemon.SdNotifyReady)
```

Comment:
I would not wait for the 5 minute reconnect. I would try to connect to Redis and the database before it notifies the ready state and cancel if it is not a retryable error. Would that be ok?

Reply:
Would be an improvement but still somewhat strange if you restart Icinga DB, it reports OK and dies after 5 minutes because it was never connected to the database.

Reply:
Without `Type = notify` we would have the same situation. To me, signaling readiness means that Icinga DB has finished parsing its configuration and is ready to do its job. The connection to the database and Redis is part of this job and I would not make the ready state dependent on it. Because if the database connection or Redis is gone during runtime, we also do not signal that we are no longer doing anything. I would rather send a `STATUS` string that we're retrying those connections.